### PR TITLE
Add annotation id to onExportAnnotationCommand

### PR DESCRIPTION
**Pull request by akwanpdf**
Added new parameter `annotations` to `onExportAnnotationCommand`. `annotations` is an array of changed annots with the following data:

- id
- page number (non-collab mode only)
- type (non-collab mode only)

### CODE REVIEW

File: android/src/main/java/com/pdftron/reactnative/views/DocumentView.java [hunk 1]
```diff
@@ -2302,13 +2302,45 @@ private void handleExportAnnotationCommand(String action, Map<Annot, Integer> ma
...
+                } catch (Exception e) {
+                    e.printStackTrace();
+                }
+
+                if (uid == null) {
+                    uid = UUID.randomUUID().toString();
+                    try {
+                        key.setUniqueID(uid);
+                    } catch (PDFNetException e) {
+                        e.printStackTrace();
```
Comment (sgong-pdftron) [inline android/src/main/java/com/pdftron/reactnative/views/DocumentView.java:2321]: - [ ] should null uid here if failed to set for the annotation

File: android/src/main/java/com/pdftron/reactnative/views/DocumentView.java [hunk 2]
```diff
@@ -2387,9 +2419,34 @@ public void onTabDocumentLoaded(String tag) {
...
+                                    } catch (Exception e) {
+                                        e.printStackTrace();
+                                    }
+
+                                    if (uid == null) {
+                                        uid = UUID.randomUUID().toString();
+                                        try {
+                                            annot.setId(uid);
+                                        } catch (Exception e) {
+                                            e.printStackTrace();
```
Comment (sgong-pdftron) [inline android/src/main/java/com/pdftron/reactnative/views/DocumentView.java:2437]: - [ ] same for this catch block

**Review by sgong-pdftron — CHANGES_REQUESTED** [review]

File: API.md [hunk 3]
```diff
@@ -1014,21 +1014,35 @@ Defines whether an annotation is selected after it is created. On iOS, this func
...
 Parameters:
 
 Name | Type | Description
 --- | --- | ---
 action | string | the action that occurred (add, delete, modify)
 xfdfCommand | string | an xfdf string containing info about the edit
+annotations | array | an array of annotation data. When collaboration is enabled data comes in the format `{id: string}`, otherwise the format is `{id: string, pageNumber: number, type: string}`. In both cases, the data represents the annotations that have been changed. Type is one of the [`Config.Tools`](./src/Config/Config.js) constants 
+
+**Known Issues** <br/> 
+On iOS, there is currently a bug that prevents the last XFDF from being retrieved when modifying annotations while collaboration mode is enabled.
```
Comment (JamesTron) [inline API.md:1029]: We should be able to fix this by pointing it to a new stable branch, right?

**Review by JamesTron — APPROVED** [review]

**Review by sgong-pdftron — APPROVED** [review]

**Comment by sgong-pdftron** [comment]
closes https://github.com/PDFTron/pdftron-react-native/issues/293

**Pull request merged**